### PR TITLE
Update Dockerfile

### PR DESCRIPTION
**Pull request by csstaub**
r: @jfrazelle @calavera @mcpherrinm @alokmenghrajani 

Updates the Dockerfile for building a Keywhiz server container.

Changes:
* Split mvn verify/install steps to cache dependency downloads
* Added entrypoint for running shaded server jar by default

Example usage for development, with persistent H2 DB in local volume:
```bash
docker volume create keywhiz-db-devel
docker run -v keywhiz-db-devel:/tmp/h2_data square/keywhiz migrate
docker run -it -v keywhiz-db-devel:/tmp/h2_data square/keywhiz server
```

### CODE REVIEW

File: docker.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    KEYWHIZ_CONFIG=server/src/main/resources/keywhiz-development.yaml
+    echo -n "${RED}"
+    echo "------------------------------------------------------------------------"
+    echo "---    No configuration file specified, using development config!    ---"
+    echo "--- Use the KEYWHIZ_CONFIG environment variable to set a config file ---" 
+    echo "------------------------------------------------------------------------"
+    echo -n "${RESET}"
+fi
+
+java -jar server/target/keywhiz-server-*-SNAPSHOT-shaded.jar $1 $KEYWHIZ_CONFIG
```
Comment (mcpherrinm) [inline docker.sh]: I don't like our use of -*- here -- it breaks if you've built two versions.
I'm not sure what the best way to fix this is, though.

**Comment by mcpherrinm** [comment]
Looks good to me.  We should investigate adding a docker test to our travis-ci config to ensure this keeps working, too.

File: docker.sh (hunk 1 above)
Comment (csstaub) [inline docker.sh]: I mean, there will only ever be one version in the image/container. For a new version we'd build a new image.

Comment (mcpherrinm) [inline docker.sh]: I guess so - but I'm not very familiar with docker.  You can't end up in a state where you have old build artifacts hanging around, accidentally included?

Comment (csstaub) [inline docker.sh]: I don't think so -- the dockerfile describes how the image is built, starting from a clean slate.

Comment (mcpherrinm) [inline docker.sh]: Alright, sounds good then.  :ship:

**Comment by calavera** [comment]
nice, looks good to me.

**Pull request merged**